### PR TITLE
Refactor Test/Build Details Page

### LINKED ISSUE #507: Populate  Hardware Test Details Page
Test Details page already exists.
We already have the route for it.

To accomplish this task you will need to point to the same component as Test Details is using right now, but it will require a different Breadcrumb, that is why we have two Routers for the same component

### PR DESCRIPTION
**Pull request by Francisco2002**
- Use breadcrumb as a prop in test and build details page
- Move logic to a general Test/Build Details component

Close #507 and #508

### CODE REVIEW

File: dashboard/src/pages/TestDetails/index.tsx
Comment (MarceloRobert) [inline dashboard/src/pages/TestDetails/index.tsx:1]: Why did you delete this file?

File: dashboard/src/components/TestDetails/TestDetails.tsx
Comment (MarceloRobert) [inline dashboard/src/components/TestDetails/TestDetails.tsx:1]: The build info link is still always redirecting to /tree (check new line 33)

Comment (Francisco2002) [reply dashboard/src/components/TestDetails/TestDetails.tsx:1]: thanks for notice this

File: dashboard/src/pages/TestDetails/index.tsx
Comment (MarceloRobert) [reply dashboard/src/pages/TestDetails/index.tsx:1]: You could have moved it to /components/TestDetails, but there was also some confusion on my part, Github doesn't show that the folder /pages/TestDetails was deleted, only the file

**Review by MarceloRobert — APPROVED** [review]
Seems great

File: dashboard/src/pages/TreeTestDetails/TreeTestDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  return (
+    <TestDetails
+      testId={testId}
+      context="tree"
+      breadcrumb={
+        <Breadcrumb className="pb-6 pt-6">
+          <BreadcrumbList>
+            <BreadcrumbItem>
+              <BreadcrumbLink to="/tree" search={searchParams}>
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeTestDetails/TreeTestDetails.tsx:28]: `previousSearchParams => { ....previousSearchParams, searchParams}`

File: dashboard/src/pages/TreeTestDetails/TreeTestDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                <FormattedMessage id="tree.path" />
+              </BreadcrumbLink>
+            </BreadcrumbItem>
+            <BreadcrumbSeparator />
+            <BreadcrumbLink
+              to={`/tree/$treeId`}
+              params={{ treeId: treeId }}
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeTestDetails/TreeTestDetails.tsx:35]: useMemo

File: dashboard/src/pages/TreeTestDetails/TreeTestDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+              search={searchParams}
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeTestDetails/TreeTestDetails.tsx:36]: previousSearchParams => { ....previousSearchParams, searchParams}

File: dashboard/src/pages/TreeBuildDetails/TreeBuildDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        <BreadcrumbItem>
+          <BreadcrumbLink to="/tree" search={searchParams}>
+            <FormattedMessage id="tree.path" />
+          </BreadcrumbLink>
+        </BreadcrumbItem>
+        <BreadcrumbSeparator />
+        <BreadcrumbLink
+          to={`/tree/$treeId`}
+          params={{ treeId: treeId }}
+          search={searchParams}
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeBuildDetails/TreeBuildDetails.tsx:57]: previousSearchParams => { ....previousSearchParams, searchParams}

File: dashboard/src/pages/TreeBuildDetails/TreeBuildDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      <BreadcrumbList>
+        <BreadcrumbItem>
+          <BreadcrumbLink to="/tree" search={searchParams}>
+            <FormattedMessage id="tree.path" />
+          </BreadcrumbLink>
+        </BreadcrumbItem>
+        <BreadcrumbSeparator />
+        <BreadcrumbLink
+          to={`/tree/$treeId`}
+          params={{ treeId: treeId }}
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeBuildDetails/TreeBuildDetails.tsx:56]: useMemo

File: dashboard/src/pages/HardwareTestDetails/HardwareTestDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  return (
+    <TestDetails
+      testId={testId}
+      context="hardware"
+      breadcrumb={
+        <Breadcrumb className="pb-6 pt-6">
+          <BreadcrumbList>
+            <BreadcrumbItem>
+              <BreadcrumbLink to="/hardware" search={searchParams}>
```
Comment (WilsonNet) [inline dashboard/src/pages/HardwareTestDetails/HardwareTestDetails.tsx:32]: previousSearchParams => { ....previousSearchParams, searchParams}
